win32/src/threadpool_win32.c

```diff
@@ -277,24 +289,133 @@ void threadpool_close(THANDLE(THREADPOOL) threadpool)
     }
 }
 
-THREADPOOL_WORK_ITEM_HANDLE threadpool_create_work_item(THANDLE(THREADPOOL) threadpool, THREADPOOL_WORK_FUNCTION work_function, void* work_function_context)
+THREADPOOL_WORK_ITEM_HANDLE threadpool_create_work_item(THANDLE(THREADPOOL) threadpool, THREADPOOL_WORK_FUNCTION work_function, PVOID work_function_context)
 {
-    (void*)threadpool;
-    (void*)work_function;
-    (void*)work_function_context;
-    return NULL;
+    THREADPOOL_WORK_ITEM_HANDLE work_item_context = NULL;
+
+    /* Codes_SRS_THREADPOOL_WIN32_01_022: [ work_function_context shall be allowed to be NULL. ]*/
+
+    if (
+        /* Codes_SRS_THREADPOOL_WIN32_05_004: [ If threadpool is NULL, threadpool_create_work_item shall fail and return a NULL value. ]*/
+        (threadpool == NULL) ||
+        /* Codes_SRS_THREADPOOL_WIN32_05_005: [ If work_function is NULL, threadpool_create_work_item shall fail and return a NULL value. ]*/
+        (work_function == NULL)
+        )
+    {
+        LogError("Invalid arguments: THANDLE(THREADPOOL) threadpool=%p, THREADPOOL_WORK_FUNCTION work_function=%p, void* work_function_context=%p",
+            threadpool, work_function, work_function_context);
+    }
+    else
+    {
+        THREADPOOL* threadpool_ptr = THANDLE_GET_T(THREADPOOL)(threadpool);
+
+        (void)InterlockedIncrement(&threadpool_ptr->pending_api_calls);
+
+        THREADPOOL_WIN32_STATE state = InterlockedAdd(&threadpool_ptr->state, 0);
+        if (state != (LONG)THREADPOOL_WIN32_STATE_OPEN)

```

Comment:
>(LONG) [](http://example.com/codeflow?start=21&length=6)

I don't think you need a cast here.

Reply:
Hi Jelani, just referenced from `threadpool_schedule_work`. Should I make these changes for all such occurrences in the file?

Reply:
Sure, I think all these types of casts are unnecessary.